ext/DifferentiationInterfaceZygoteExt.jl

```diff
@@ -1,7 +1,51 @@
 module DifferentiationInterfaceZygoteExt
 
-using DifferentiationInterface
+using DifferentiationInterface: ChainRulesReverseBackend
+import DifferentiationInterface as DI
 using DocStringExtensions
-using Zygote
+using Zygote: ZygoteRuleConfig, gradient, jacobian, withgradient, withjacobian
+
+const ZygoteBackend = ChainRulesReverseBackend{<:ZygoteRuleConfig}
+
+## Special cases
+
+"""
+$(TYPEDSIGNATURES)
+"""
+function DI.value_and_gradient(::ZygoteBackend, f, x::AbstractArray)
+    res = withgradient(f, x)
+    return res.val, only(res.grad)
+end
+
+"""
+$(TYPEDSIGNATURES)
+"""
+function DI.value_and_gradient!(
+    grad::AbstractArray, backend::ZygoteBackend, f, x::AbstractArray
+)
+    y, new_grad = DI.value_and_gradient(backend, f, x)
+    grad .= new_grad
+    return y, grad
+end
+
+"""
+$(TYPEDSIGNATURES)
+"""
+function DI.value_and_jacobian(::ZygoteBackend, f, x::AbstractArray)
+    y = f(x)
+    jac = jacobian(f, x)
+    return y, only(jac)
+end
+
+"""
+$(TYPEDSIGNATURES)
+"""
+function DI.value_and_jacobian!(
+    jac::AbstractMatrix, backend::ZygoteBackend, f, x::AbstractArray
+)
+    y, new_jac = DI.value_and_jacobian(backend, f, x)
+    jac .= new_jac
+    return y, jac
+end
```

Comment:
Zygote's `withjacobian` would be more efficient as it returns the primal value as well:
https://fluxml.ai/Zygote.jl/dev/utils/#Zygote.withjacobian

Reply:
Yes but it returns a flattened version, as I unfortunately discovered. So I need to run f anyway to get the correct shape

Reply:
Is the primal output of `withjacobian` also flattened?

Reply:
Yes that is what I mean

Reply:
```julia
julia> Zygote.withjacobian(identity, rand(2, 3)).val
6-element Vector{Float64}:
 0.42579108690107337
 0.9756201877495323
 0.3391154024834867
 0.8593637011392009
 0.6974918525763482
 0.9470770391746074
```

Reply:
https://github.com/FluxML/Zygote.jl/issues/1506